pkgs/by-name/li/lightning-terminal/package.nix

```diff
@@ -75,24 +94,65 @@ buildGoModule rec {
     versionCheckHook
   ];
 
-  lightning-app = mkYarnPackage {
+  passthru.tests.litd-app = testers.runCommand {
+    name = "test-litd-app";
+    nativeBuildInputs = [
+      curl
+      lightning-terminal
+    ];
+    script = ''
+      litd \
+        --uipassword=12345678 \
+        --insecure-httplisten=127.0.0.1:8080 \
+        --httpslisten= &
+      sleep 2
+      GETindexHTTPCode=$(curl -o /dev/null -w "%{http_code}" -Lvs 127.0.0.1:8080/index.html)
+      if [ "$GETindexHTTPCode" = 200 ]; then
+        touch $out
+      fi
+    '';
+  };
+
+  passthru.tests.version = testers.testVersion {
+    package = lightning-terminal;
+    command = "litd --version";
+    version = "litd version ${version} commit=v${version}";
+  };
+
```

Comment:
There's a shortcut: [`versionCheckHook`](https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook) (see a little bit above in the code).
```suggestion
```